tests/virt/vmfakelib.py

```diff
@@ -51,6 +51,9 @@ def appropriateDevice(self, guid, vmid):
     def inappropriateDevices(self, ident):
         pass
 
+    def prune_bitmaps(self, sdUUID, imgUUID, volUUID, baseUUID):
+        pass
```

Comment:
This should use `@mayfail` so we can simulate errors and test merge behavior on failures.

It should also record the fact that this was called and the called arguments, we have
a `@recorded` decorator for this, see how it is used. This will allow verifying that we
prone bitmaps when starting merge for internal merge, and not pruning bitmaps for
active layer merge or internal merge to raw base.

Check how livemerge_test.py verify the flows. We need to add some more asserts in the
existing tests.

Reply:
I saw how `@maybefail` works, is a handy wrapper!

But in this case I think is better fit to use the `response.error_raw` if we want to simulate errors, something like what we do in `getVolumeSize` below, to have similar code to this one:
```python
    @recorded
    def prune_bitmaps(self, sdUUID, imgUUID, volUUID, baseUUID):
        if "prune_bitmaps" in self.errors:
            error = self.errors["prune_bitmaps"]
            return response.error_raw(error.code, error.msg)
        return response.success(result=None)
```

Reply:
VM.prune_bitmaps() should not return status code! It should raise on errors. Its purpose is to covert the (horrible) self.irs.cif.prune_bitmaps() call returning status codes to proper call raising on errors.

Reply:
It does raise on errors. This `prune_bitmaps` is in the fake IRS class, not in the VM.

So it returns a status code that is processed by `VM.prune_bitmaps()`, and then raise the error.

The fact that both have the same name may cause some confusion.

Reply:
The fake IRS should be wrapped with a Dispatcher so it behaves like the real cif.irs object.

But if the current code is not using a dispatcher and try to behave like a wrapped hsm
and return status codes, you should do the same.